src/main/scala/ai/privado/tagger/sink/api/InferenceAPIEndpointTagger.scala

```diff
@@ -21,38 +23,51 @@ class InferenceAPIEndpointTagger(cpg: Cpg, ruleCache: RuleCache) extends Privado
   private val logger = LoggerFactory.getLogger(getClass)
 
   override def generateParts(): Array[RuleInfo] =
-    ruleCache.getRule.inferences.filter(_.catLevelTwo.equals(Constants.apiEndpoint)).toArray
+    Try(ruleCache.getRule.inferences.filter(_.catLevelTwo.equals(Constants.apiEndpoint)).toArray) match
```

Comment:
Looking at the contents of the generateParts(). Looks like there is less possibility of error getting thrown from this place.